pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java

```diff
@@ -691,7 +696,120 @@ public Reader getCharacterStream(String parameterName) throws SQLException {
   }
 
   public <T> T getObject(int parameterIndex, Class<T> type) throws SQLException {
-    if (type == ResultSet.class) {
+    if (type == null) {
+      throw new SQLException("type is null");
+    }
+    checkIndex(parameterIndex);
+    int sqlType = functionReturnType[parameterIndex - 1];
+    if (type == BigDecimal.class) {
+      if (sqlType == Types.NUMERIC || sqlType == Types.DECIMAL) {
+        return type.cast(getBigDecimal(parameterIndex));
+      } else {
+        throw new SQLException("conversion to " + type + " from " + sqlType + " not supported");
+      }
+    } else if (type == String.class) {
+      if (sqlType == Types.CHAR || sqlType == Types.VARCHAR) {
+        return type.cast(getString(parameterIndex));
+      } else {
+        throw new SQLException("conversion to " + type + " from " + sqlType + " not supported");
+      }
+    } else if (type == Boolean.class) {
+      if (sqlType == Types.BOOLEAN || sqlType == Types.BIT) {
```

Comment:
Without checking any other drivers `BitSet` would be my immediate choice as well. Does anybody know what SQL Server does? If both mssql-jdbc and connector-j return `byte[]` an argument can be made that `#getObject(int)` should return `byte[]` for compatibility reasons and `BitSet` could still be supported with `#getObject(int, Class)`. Again I would like discuss this in a different issue.